source-mongodb/change_stream.go

```diff
@@ -0,0 +1,404 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/segmentio/encoding/json"
+	log "github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"golang.org/x/sync/errgroup"
+)
+
+// changeStream associates a *mongo.ChangeStream with which database it is for.
+type changeStream struct {
+	ms *mongo.ChangeStream
+	db string
+}
+
+// makeEventDocument processes a change event and returns a document if the change event is for a
+// tracked collection and operation type. The binding index is also returned if a document is
+// available from the change event. If no document is applicable to the change event, the boolean
+// output will be false.
+func makeEventDocument(ev changeEvent, collectionBindingIndex map[string]bindingInfo) (map[string]any, int, bool) {
```

Comment:
I went back and fourth a bit on this. I personally like the explicit boolean but it may be more idiomatic to do `nil` instead, so I'll update it to be that.